diff --git dependencies/jedi/_compatibility.py dependencies/jedi/_compatibility.py
index 065fde89..85c7ab8d 100644
--- dependencies/jedi/_compatibility.py
+++ dependencies/jedi/_compatibility.py
@@ -198,7 +198,7 @@ def find_module_pre_py3(string, path=None, full_name=None, is_global_search=True
     raise ImportError("No module named {}".format(string))


-find_module = find_module_py34 if is_py3 else find_module_pre_py3
+find_module = find_module_py33 if is_py3 else find_module_pre_py3
 find_module.__doc__ = """
 Provides information about a module.

@@ -522,7 +522,7 @@ def _python2_dct_keys_to_unicode(data):

 def pickle_dump(data, file, protocol):
     try:
-        if not is_py3:
+        if py_version < 34:
             data = _python2_dct_keys_to_unicode(data)
         pickle.dump(data, file, protocol)
         # On Python 3.3 flush throws sometimes an error even though the writing
@@ -640,7 +640,7 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
     return None


-if not is_py3:
+if py_version < 34:
     # Simplified backport of Python 3 weakref.finalize:
     # https://github.com/python/cpython/blob/ded4737989316653469763230036b04513cb62b3/Lib/weakref.py#L502-L662
     class finalize(object):
diff --git dependencies/jedi/inference/compiled/access.py dependencies/jedi/inference/compiled/access.py
index de762c36..cd684673 100644
--- dependencies/jedi/inference/compiled/access.py
+++ dependencies/jedi/inference/compiled/access.py
@@ -34,9 +34,11 @@ if is_py3:
     NOT_CLASS_TYPES += (
         types.MappingProxyType,
         types.SimpleNamespace,
-        types.DynamicClassAttribute,
     )

+    if py_version >= 34:
+        NOT_CLASS_TYPES += (types.DynamicClassAttribute,)
+

 # Those types don't exist in typing.
 MethodDescriptorType = type(str.replace)
