diff --git dependencies/jedi/_compatibility.py dependencies/jedi/_compatibility.py
index 709cd518..f5a5ad4a 100644
--- dependencies/jedi/_compatibility.py
+++ dependencies/jedi/_compatibility.py
@@ -640,7 +640,7 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
     return None
 
 
-if not is_py3:
+if py_version < 34:
     # Simplified backport of Python 3 weakref.finalize:
     # https://github.com/python/cpython/blodependencies/ded4737989316653469763230036b04513cb62b3/Lidependencies/weakref.py#L502-L662
     class finalize(object):
diff --git dependencies/jedi/inference/compiled/access.py dependencies/jedi/inference/compiled/access.py
index f47ae773..cc1151f5 100644
--- dependencies/jedi/inference/compiled/access.py
+++ dependencies/jedi/inference/compiled/access.py
@@ -35,7 +35,6 @@ if is_py3:
     NOT_CLASS_TYPES += (
         types.MappingProxyType,
         types.SimpleNamespace,
-        types.DynamicClassAttribute,
     )
