diff --git dependencies/jedi/_compatibility.py dependencies/jedi/_compatibility.py
index 9b5edc8..81b844a 100644
--- dependencies/jedi/_compatibility.py
+++ dependencies/jedi/_compatibility.py
@@ -150,6 +150,7 @@ def find_module_pre_py34(string, path=None, full_name=None, is_global_search=Tru
 
 
 find_module = find_module_py34 if is_py3 else find_module_pre_py34
+find_module = find_module_py33 if py_version == 33 else find_module
 find_module.__doc__ = """
 Provides information about a module.
 
diff --git dependencies/jedi/evaluate/compiled/access.py dependencies/jedi/evaluate/compiled/access.py
index ab1d354..7fd5811 100644
--- dependencies/jedi/evaluate/compiled/access.py
+++ dependencies/jedi/evaluate/compiled/access.py
@@ -5,7 +5,7 @@ from textwrap import dedent
 import operator as op
 from collections import namedtuple
 
-from jedi._compatibility import unicode, is_py3, builtins, \
+from jedi._compatibility import unicode, is_py3, py_version, builtins, \
     py_version, force_unicode, print_to_stderr
 from jedi.evaluate.compiled.getattr_static import getattr_static
 
@@ -32,9 +32,11 @@ if is_py3:
     NOT_CLASS_TYPES += (
         types.MappingProxyType,
         types.SimpleNamespace,
-        types.DynamicClassAttribute,
     )
 
+    if not py_version == 33:
+        NOT_CLASS_TYPES += (types.DynamicClassAttribute,)
+
 
 # Those types don't exist in typing.
 MethodDescriptorType = type(str.replace)
